src/code/z_vr_box_draw.c

```diff
@@ -1,5 +1,7 @@
 #include "global.h"
 
+#define SKYBOX_PALETTES ((u16(*)[256])skyboxCtx->staticSegments[2])
```

Comment:
I'm not sure this macro really improves things tbh. I feel like it hides important details, and it relies on a specific local variable which we're tring to avoid nowadays.

The previous pointer arithmetic seemed fine to me but we probably need more opinions on this.

Reply:
I like the macro much better, I think it is a lot more readable this way

Would you settle on `SKYBOX_PALETTES[n]` -> `SKYBOX_PALETTES(skyboxCtx)[3]` ?

But, does `staticSegments` have to be an array? I know that's what debug strings say it was, but if it can match as individual struct members it would be better

Reply:
I personally think hiding important details for readability is a bad idea, but maybe that's just me.

If we really don't want pointer arithmetic for this specific thing, then a decent compromise would be to change `staticSegments[2]` to an individual member (while keeping the other 2 pointers as a generic staticSegments array).

Reply:
Im kind of in the middle on this. I understand wanting to make it more readable, but also feel the same about hiding the details.
I think in this case it doesnt really help with much, there isnt any ambiguity of what it is. They are being directly used in tlut commands, so itd be very clear at first glance that they are palletes.

Reply:
```c
    /* 0x128 */ void* staticSegments[3];
```
also matches as
```c
    /* 0x128 */ void* staticSegments0;
    /* 0x12C */ void* staticSegments1;
    /* 0x130 */ u16 (*skyboxPalettes)[256];
```
would that be fine or do we really need to go with what the debug string says?

Reply:
id be fine with this personally

Reply:
I would be okay with the variant of this I suggested above, where the first 2 are kept as an array (which could easily have more entries in a modding scenario)